cql/CollateManagementData.cql

```diff
@@ -387,6 +387,27 @@ define Cin2orCin3Biopsies:
 define MostRecentCin2orCin3Biopsy:
   First(Cin2orCin3Biopsies)
 
+define Cin3Biopsies:
```

Comment:
It seems like these functions repeat a lot of what is defined in ManageRareAbnormality.cql. For example, it seems like `CIN2Biopsies` is essentially the same as `Rare.HistologyInterpretedAsCin2`. I know there are some subtle nuances, like the difference between "the most recent CIN2 histology result" and "the most recent histology result is CIN2", so I'm not entirely sure if this logic was intended to live as it's own logic that is separate from the ManageRareAbnormality.cql file, but it seems like some if not most of what is here can be consolidated with what lives over there.

Reply:
They are different functions and have different syntactical usage in the CQL. We could consider where such definitions belong and whether we need to consolidate. We also need to consider the need to prefix all these references if moved (e.g. HistologyInterpretedAsCin2 becomes Collate.HistologyInterpretedAsCin2 in usage)

Reply:
I see what you're saying. It appears these functions here:
- Take a subset of SortedBiopsyReports which includes just biopsy reports of a given type (ex. "Cin3Biopsies")
- Take the most recent of all the biopsies that have a result of that type (ex. MostRecentCin3Biopsy)

Whereas there seem to be two primary types of expressions for histologies over in ManageRareAbnormality.cql:
1. "HistologyInterpretedAsCin3" - takes the list of _all_ biopsies, and returns the most recent one if it happens to have a result of CIN3.
2. "AnyHistologyInterpretedAsCin3" - returns a boolean: 'true' if any biopsy result has ever been CIN3, `false` if otherwise.

I see now the distinction, but I still think there could be some room for consolidation here. Some thoughts:
- I think we should consider creating a standard dictionary for how we label these type of lab results, and maybe consider consolidating all of the expressions around test results into the same file, with better commenting to express what is meant by each expression, if the expression name could be interpreted in multiple ways
- Some of the expressions defined here could be used as the base to many of the expressions in ManageRareAbnormality.cql. For example, `AnyHistologyInterpretedAsCin3` is essentially just the boolean version of `MostRecentCin3Biopsy`. `MostRecentCin2BiopsyInPast2Years` is essentially just `MostRecentCin2Biopsy` but with one extra line to institute a 2-year lookback. I building expressions from existing expressions like this could make things more efficient and repeatable and could make it clearer to track down how things are defined.

These things aren't actually affecting the accuracy of the sections modified in this PR, so I would be fine with these things addressed by a follow-up PR around refactoring how we query lab tests, maybe after we discuss some of the thoughts above. @mrnosal If you do plan on leaving this unchanged for this PR, would you mind opening a Jira ticket so that we can track the thoughts I outlined above, as well as any follow-up thoughts you have?

Reply:
I'm addressing this in the HSIL branch for PR120 - I'll change the names of the expressions.

Reply:
- Names were changed in PR 120 by commit https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/pull/120/commits/d10. This addresses the "standard dictionary" idea brought up here.
- As for consolidating and refactoring, I realize that this may be a bigger undertaking than just what is involved in this PR, so I created a Jira ticket at the bottom of the backlog to address it: [CCSMCDS-81](https://jira.mitre.org/browse/CCSMCDS-81).